src/server/mod.rs

```diff
@@ -28,6 +28,7 @@ pub struct TlsConfig {
 /// Returns a future that runs a server until it receives a SIGINT to stop. If optional TLS
 /// configuration is given, the server will be configured to use TLS. Otherwise it will use plain
 /// HTTP
+// TODO: Change this to a builder instead as this is too many arguments
```

Comment:
Please note: I am working on a follow up PR for this TODO. Didn't want to mix it with this PR

Reply:
So making a builder that works with generics is fairly impossible because once you choose a type, it needs to stay that type the whole time. Just going to remove this todo and ignore the clippy lint

Reply:
To clarify, all 5 of the generic parameters would need to be provided up front. We'd only be removing 3 things. So it probably isn't worth it